packages/api-client/src/views/Request/ResponseSection/ResponseSection.vue

```diff
@@ -59,9 +59,29 @@ const activeSection = ref<ActiveSections>('All')
 
 /** Threshold for virtualizing response bodies in bytes */
 const VIRTUALIZATION_THRESHOLD = 200_000
-const shouldVirtualize = computed(
-  () => (props.response?.size ?? 0) > VIRTUALIZATION_THRESHOLD,
-)
+const shouldVirtualize = computed(() => {
+  if (!props.response) return false
+
+  // Get content type from headers
+  const contentType =
+    props.response.headers?.['content-type'] ||
+    props.response.headers?.['Content-Type']
+
+  // If no content type or response size is small, don't virtualize
+  if (!contentType || (props.response.size ?? 0) <= VIRTUALIZATION_THRESHOLD) {
+    return false
+  }
+
+  // Check if content type is text-based
+  const isTextBased =
+    contentType.includes('text/') ||
+    contentType.includes('application/json') ||
+    contentType.includes('application/xml') ||
+    contentType.includes('application/yaml') ||
+    contentType.includes('application/javascript')
```

Comment:
This works for me but I did a small research and found around 48 text-based content types, I'll share it here in case it helps:

### General Text Content Types
1. `text/plain` - Plain text
2. `text/html` - HTML documents
3. `text/css` - Cascading Style Sheets (CSS)
4. `text/javascript` - JavaScript (deprecated in favor of `application/javascript`)
5. `text/xml` - XML format (deprecated in favor of `application/xml`)
6. `text/csv` - CSV (Comma-Separated Values)
7. `text/markdown` - Markdown format
8. `text/richtext` - Rich Text Format (deprecated)
9. `text/vcard` - vCard format
10. `text/calendar` - iCalendar format

### Application Text Content Types
11. `application/json` - JSON format
12. `application/xml` - XML format
13. `application/xhtml+xml` - XHTML format
14. `application/x-www-form-urlencoded` - URL-encoded form data
15. `application/javascript` - JavaScript
16. `application/ecmascript` - ECMAScript
17. `application/ld+json` - JSON-LD format
18. `application/rtf` - Rich Text Format
19. `application/atom+xml` - Atom feeds
20. `application/rss+xml` - RSS feeds
21. `application/vnd.api+json` - JSON:API format
22. `application/xml-dtd` - Document Type Definitions (DTD)
23. `application/sparql-query` - SPARQL queries
24. `application/sparql-results+xml` - SPARQL query results in XML
25. `application/x-yaml` - YAML format
26. `application/problem+json` - Problem details in JSON format
27. `application/problem+xml` - Problem details in XML format

### Specialized Text Content Types
28. `application/pgp-signature` - PGP signatures
29. `application/pgp-encrypted` - PGP encrypted data
30. `application/pgp-keys` - PGP keys
31. `application/x-sh` - Shell script
32. `application/x-csh` - C shell script
33. `application/x-perl` - Perl script
34. `application/x-python` - Python script
35. `application/x-ruby` - Ruby script
36. `application/x-php` - PHP script
37. `application/x-java-source` - Java source code
38. `application/x-tcl` - Tcl script
39. `application/x-lua` - Lua script
40. `application/x-httpd-php` - PHP source code
41. `application/x-httpd-eruby` - eRuby script
42. `application/x-latex` - LaTeX document

### Other Text-Based Content Types
43. `application/vnd.oasis.opendocument.text` - OpenDocument Text
44. `application/vnd.oasis.opendocument.text-template` - OpenDocument Text Template
45. `application/vnd.openxmlformats-officedocument.wordprocessingml.document` - Microsoft Word document (OpenXML)
46. `application/vnd.openxmlformats-officedocument.wordprocessingml.template` - Microsoft Word template (OpenXML)
47. `application/vnd.ms-word.document.macroEnabled.12` - Microsoft Word macro-enabled document
48. `application/vnd.ms-word.template.macroEnabled.12` - Microsoft Word macro-enabled template

----
to me things like `application/ld+json` or `application/problem+json` are important ones. although I'm not sure what Virtualization does exactly.

Reply:
added them :) appreciate this!